### DISCUSSION TITLE
Filter does not work

### DISCUSSION THREAD

**Discussion by JivanaLipsewers** [neuefische/questions#650] [Web]
hello all, unfortunately I can not find my error. 

I have three categories of questions "About me", "About you", "About us", which can be selected via the filter "Select category". If you don't select any checkbox, all questions from each category should be displayed (status quo), but if you click one or/and a second category specifically, only those questions from that category should be displayed.

Can someone take a look at the components > filter > index and the pages > app.js? Thanks! :)

-> [Capstone PR](https://github.com/JivanaLipsewers/Capstone-90-Fragen-die-verbinden/pull/14)

**Comment by ahohnsen** [comment]
Hey Jivana,

You described how your filter should be working in the end. Can you also specify what exactly is not working/which error you are getting?

This will help answering your question. 🙏🏼 
Thank you!

**Reply by JivanaLipsewers** [reply]
Hey Andrea, 

oh sure! My checkboxes in the filter do not respond. If I click on a specific category, then nothing changes and the category is not filtered, but everything is as before. 

Maybe you have a tip for me? 
Thanks!

**Reply by ahohnsen** [reply]
Hey Jivana,

When the user chooses a category, the category is saved correctly in `selectedCategories`.
You are just missing the next step: You need to filter all questions in order to only show the once, which are matching the chosen category/categories. 
You can use the method `filter` on your `questions` state and check if the category of the question is included in your `selectedCategories` state. The `filter` method will return a new array with only matching questions, which needs to be saved in a variable (here: `filteredQuestions`). You can use a ternary operator to only filter the questions, if there is actually a category selected (`selectedCategories.length ?` ), otherwise all questions will be saved in `filteredQuestions`.

```jsx
const filteredQuestions = selectedCategories.length
    ? questions.filter((question) =>
        selectedCategories.includes(question.category)
      )
    : questions;
```

The variable `filteredQuestions` will then be used to determine the `currentQuestion` and the `currentQuestionIndex`.

```jsx 
const currentQuestion = filteredQuestions[currentQuestionIndex];
```

```jsx 
  function handleNextQuestion() {
    if (currentQuestionIndex + 1 < filteredQuestions.length) {
      setCurrentQuestionIndex(currentQuestionIndex + 1);
    }
  }
```

Additionally, I strongly recommend you to have a look at your `_app.js` again. This file should actually return something like this:

```jsx
return (
    <>
      <GlobalStyle />
      <Component
        {...pageProps}
      />
    </>
  );
```
What you return right now in your `_app.js `should actually be in your `index.js`.

I hope that helps ✌🏼
